File: Packets/src/main/java/me/ImSpooks/core/packets/collection/database/PacketRequestDataResponse.java
package me.ImSpooks.core.packets.collection.database;

import lombok.Getter;
import me.ImSpooks.core.helpers.Global;
import me.ImSpooks.core.packets.collection.GlobalPacket;
import me.ImSpooks.core.packets.init.Packet;
import me.ImSpooks.core.packets.init.channels.WrappedInputStream;
import me.ImSpooks.core.packets.init.channels.WrappedOutputStream;
import org.bson.Document;

import java.io.IOException;

/**
 * Created by Nick on 09 okt. 2019.
 * Copyright © ImSpooks
 */
@GlobalPacket
public class PacketRequestDataResponse extends Packet {

    @Getter private Document document;

    public PacketRequestDataResponse(Document document) {
        this.document = document;
    }

    public PacketRequestDataResponse() {
    }

    @Override
    public void send(WrappedOutputStream out) throws IOException {
        out.writeString(Global.GSON.toJson(this.document));
    }

    @Override
    public void receive(WrappedInputStream in) throws IOException {
        this.document = Global.GSON.fromJson(in.readString(), Document.class);
    }
}

File: ClientBridge/src/main/php/client/packets/collection/database/PacketRequestDataResponse.php
<?php
/**
 * Created by Nick on 25 okt 2019.
 * Copyright © ImSpooks
 */

namespace client\packets\collection\database;

use client\packets\init\Packet;
use client\packets\init\channels\WrappedInputStream;
use client\packets\init\channels\WrappedOutputStream;

class PacketRequestDataResponse extends Packet {

    /** @var array */ 	private $document;

    public function __construct(array $document = []) {
        $this->document = $document;
    }

    public function receive(WrappedInputStream $in) {
        $this->document = json_decode($in->readString());
    }

    public function send(WrappedOutputStream $out) {
        $out->writeString(json_encode($this->document));
    }

    public function getDocument(): array {
        return $this->document;
    }

}
